tests/unit/test_svs.cpp

```diff
@@ -0,0 +1,1834 @@
+/* TODO: change the copyright here */
+
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "gtest/gtest.h"
+#include "VecSim/vec_sim.h"
+#include "unit_test_utils.h"
+#include "VecSim/algorithms/svs/svs.h"
+#include "cpu_features_macros.h"
+#include <cmath>
+
+template <typename index_type_t>
+class SVSTest : public ::testing::Test {
+public:
+    using data_t = typename index_type_t::data_t;
+
+protected:
+    VecSimIndex *CreateNewIndex(SVSParams &params) {
+        params.quantBits = index_type_t::get_quant_bits();
+        params.type = index_type_t::get_index_type();
+        VecSimParams index_params = CreateParams(params);
+        return VecSimIndex_New(&index_params);
+    }
+
+    SVSIndexBase *CastToSVS(VecSimIndex *index) {
+        auto indexBase = dynamic_cast<SVSIndexBase *>(index);
+        assert(indexBase != nullptr);
+        return indexBase;
+    }
+};
+
+// TEST_DATA_T and TEST_DIST_T are defined in test_utils.h
+
+template <VecSimType type, typename DataType, VecSimQuantBits quantBits>
+struct SVSIndexType {
+    static constexpr VecSimType get_index_type() { return type; }
+    static constexpr VecSimQuantBits get_quant_bits() { return quantBits; }
+    typedef DataType data_t;
+};
+
+// clang-format off
+using SVSDataTypeSet = ::testing::Types<SVSIndexType<VecSimType_FLOAT32, float, VecSimQuant_NONE>
+                                    //    ,SVSIndexType<VecSimType_FLOAT32, float, VecSimQuant_8>
```

Comment:
why is it disabled?

Reply:
Fixed - these cases now enabled for LVQ only.